### ISSUE TITLE
Marathon Basic Auth/TLS Broken

### ISSUE THREAD

**Issue by arodd** [containous/traefik#365]
Neither of... basic authentication or the TLS config appears to function correctly when testing against a marathon instance with both basic auth and an invalid certificate hostname.  Eliminating the SSL portion from the test, the basic authentication portion still doesn't seem to work.

https://github.com/containous/traefik/commit/c46ffed846af0e3d7a0ea1feb016831710e2af5a#diff-fca3fabc700de27831fb573ad618a81d
https://github.com/containous/traefik/commit/9a9c8e57095f443cdc805849c63b1846f1190cad#diff-fca3fabc700de27831fb573ad618a81d

**Comment by emilevauge** [comment]
Hi @arodd, do you have any logs?

**Comment by arodd** [comment]
http://pastebin.com/2d9AmaJx

I've tried just about every combination of capitalization on the marathon.Basic marathon.basic httpBasicUserAuth/HTTPBasicUserAuth/HttpBasicUserAuth/InsecureSkipVerify/marathon.tls/marathon.TLS/etc.  I settled on using what was directly in the provider source code for marathon.go which is shown in the pastebin above.  Basically the first results are only of it failing basic auth, because i manually added a record in the root sub-domain where the cert is valid, the second example is using the service discovery url which doesn't currently have a valid cert name match and insecureskipverify doesn't seem to take effect.
